sqlmesh/core/model/definition.py

```diff
@@ -727,7 +727,13 @@ def data_hash(self) -> str:
     @property
     def _data_hash_values(self) -> t.List[str]:
         data = [
-            str(self.sorted_python_env),
+            str(  # Exclude metadata only macro funcs
+                [
+                    (k, v)
+                    for k, v in self.sorted_python_env
+                    if isinstance(v, Executable) and not v.is_metadata
```

Comment:
Shouldn't this be:
```
if not isinstance(v, Executable) or not v.is_metadata
```

Reply:
Actually, when it's ever non `Executable`?

Reply:
You're right. I should've taken a closer look at the type annotation of python_env.